app/Http/Controllers/API/StaffingController.php

```diff
@@ -0,0 +1,219 @@
+        ], 200);
+    }
+
+    public function update(Request $request, Staffing $staffing)
+    {
+        $data = $request->validate([
+            'description' => 'nullable|string',
+            'message_id' => 'nullable|integer',
+            'section_1_title' => 'nullable|string',
+            'section_2_title' => 'nullable|string',
+            'section_3_title' => 'nullable|string',
+            'section_4_title' => 'nullable|string',
+        ]);
+
+        foreach ($staffing->positions as $position) {
+            if ($position->discord_user || $position->booking_id)
+            {
+                return response()->json(['error' => 'Staffing cannot be updated because it has bookings.'], 500);
+            }
+        }
+
+        $updateData = array_filter($data, fn($value) => !is_null($value));
+
+        if (!empty($updateData)) {
+            $staffing->update($updateData);
+
+            $response = StaffingHelper::updateDiscordMessage($staffing);
+
+            if (!$response) {
+                return response()->json(['error' => 'Failed to update Discord message'], 500);
+            }
+
+            return response()->json([
+                'message' => 'Staffing updated successfully',
+            ]);
+        }
+
+        return response()->json([
+            'error' => 'No valid data provided for update',
+        ], 422);
+    }
+
+    public function reset(Staffing $staffing)
+    {
+        if (StaffingHelper::resetStaffing($staffing)) {
+            $response = StaffingHelper::updateDiscordMessage($staffing);
+
+            if (!$response) {
+                return response()->json(['error' => 'Failed to update Discord message'], 500);
+            }
+
+            return response()->json([
+                'message' => 'Staffing reset successfully',
+            ], 200);
+        }
+
+        return response()->json([
+            'error' => 'Failed to reset staffing. No valid parent or future child event found.',
+        ], 500);
+    }
+
+    /**
+     * Remove the specified resource from storage.
+     */
+    public function destroy(Request $request) 
+    {
+        $validated = $request->validate([
+            'discord_user_id' => 'required|integer',
+            'message_id' => 'required|integer',
+            'position' => 'nullable|exists:positions,callsign',
+            'section' => 'nullable|integer|between:1,4',
+        ]);
+
+        $staffing = Staffing::where('message_id', $validated['message_id'])->first();
+        
+        if (!$staffing) {
+            return response()->json(['error' => 'Staffing not found'], 404);
+        }
+
+        $positions = $staffing->positions()
+        ->when(isset($validated['position']), fn($query) => $query->where('callsign', $validated['position']))
+        ->when(isset($validated['section']), fn($query) => $query->where('section', $validated['section']))
+        ->where('discord_user', $validated['discord_user_id'])
+        ->get();
```

Comment:
question: looking at this, I'm guessing the events project doesn't use laravel's pint?